resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -102,9 +105,17 @@
     final AtomicReferenceArray<DnsQueryContext> promises = new AtomicReferenceArray<DnsQueryContext>(65536);
 
     /**
-     * The cache for {@link #query(DnsQuestion)}
+     * Cache for {@link #doResolve(InetSocketAddress, Promise)} and {@link #doResolveAll(InetSocketAddress, Promise)}.
      */
-    final ConcurrentMap<DnsQuestion, DnsCacheEntry> queryCache = PlatformDependent.newConcurrentHashMap();
+    final ConcurrentMap<String, List<DnsCacheEntry>> resolveCache = PlatformDependent.newConcurrentHashMap();
+
+    private final FastThreadLocal<Iterator<InetSocketAddress>> nameServerAddrIterator =
+            new FastThreadLocal<Iterator<InetSocketAddress>>() {
+                @Override
+                protected Iterator<InetSocketAddress> initialValue() throws Exception {
+                    return nameServerAddresses.iterator();
```

Comment:
huh... but iterator() is only called one time here which is once the FastTraeadLocal is initialised. So it is the same iterator()